src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
@textshell about https://github.com/borgbackup/borg/pull/2255#discussion_r105570618

I guess old behaviour was also bad somehow. it just sent out rollback (hopefully) and then blew up when it fell over the async exception of some unrelated command.

so it didn't really get the response for rollback (== whether it was done on the remote end). maybe it even killed the whole thing before it was even able to get to remote / being processed by remote?

(copied from irc)